npm/private/test/parse_pnpm_lock_tests.bzl

```diff
@@ -46,34 +46,92 @@ expected_packages = {
     },
 }
 
+expected_imports_injected = {
+    ".": {
+        "dependencies": {},
+        "dev_dependencies": {},
+        "optional_dependencies": {},
+    },
+    "packages/a": {
+        "dependencies": {
+            "b": "file:packages/b_typescript_5.6.2",
+        },
+        "dev_dependencies": {},
+        "optional_dependencies": {},
+    },
+    "packages/b": {
+        "dependencies": {
+            "typescript": "5.6.2",
+        },
+        "dev_dependencies": {},
+        "optional_dependencies": {},
+    },
+}
+expected_packages_injected = {
+    "file:packages/b_typescript_5.6.2": {
+        "id": "file:packages/b",
+        "name": "b",
+        "dependencies": {
+            "typescript": "5.6.2",
+        },
+        "optional_dependencies": {},
+        "dev": False,
+        "has_bin": False,
+        "optional": False,
+        "requires_build": False,
+        "version": "file:packages/b_typescript_5.6.2",
+        "friendly_version": "file:packages/b_typescript_5.6.2",
+        "resolution": {
+            "directory": "packages/b",
+            "type": "directory",
+        },
+    },
+    "typescript@5.6.2": {
+        "id": None,
+        "name": "typescript",
+        "dependencies": {},
+        "optional_dependencies": {},
+        "dev": False,
+        "has_bin": True,
+        "optional": False,
+        "requires_build": False,
+        "version": "5.6.2",
+        "friendly_version": "5.6.2",
+        "resolution": {
+            "integrity": "sha512-NW8ByodCSNCwZeghjN3o+JX5OFH0Ojg6sadjEKY4huZ52TqbJTJnDo5+Tw98lSy63NZvi4n+ez5m2u5d4PkZyw==",
+        },
+    },
+}
+
+# Example: https://github.com/pnpm/pnpm/blob/0672517f694da62dff7c33b9e723fbfb036eaefa/pnpm-lock.yaml
 def _parse_lockfile_v5_test_impl(ctx):
     env = unittest.begin(ctx)
 
-    parsed_json = pnpm.parse_pnpm_lock_json("""\
-{
-  "lockfileVersion": 5.4,
-  "specifiers": {
-    "@aspect-test/a": "5.0.0"
-  },
-  "dependencies": {
-    "@aspect-test/a": "5.0.0"
-  },
-  "packages": {
-    "/@aspect-test/a/5.0.0": {
-      "resolution": {
-        "integrity": "sha512-t/lwpVXG/jmxTotGEsmjwuihC2Lvz/Iqt63o78SI3O5XallxtFp5j2WM2M6HwkFiii9I42KdlAF8B3plZMz0Fw=="
-      },
-      "hasBin": true,
-      "dependencies": {
-        "@aspect-test/b": "5.0.0",
-        "@aspect-test/c": "1.0.0",
-        "@aspect-test/d": "2.0.0_@aspect-test+c@1.0.0"
-      },
-      "dev": false
-    }
-  }
-}
-""")
+    parsed_json = pnpm.parse_pnpm_lock_json(json.encode({
```

Comment:
Why switch to JSON? When I'm writing these tests I always generate a pnpm-lock and then paste it into here, converting to json sounds like extra work when writing tests?

Reply:
Happy to switch this back.

I've forgotten the exact reason I did this, my guess is the syntax highlighting just made it easier to scan when comparing inputs to outputs. Could also just be habit, in a few internal Rust projects we use a `json!` macro so the data is expressed in the "host" language, meaning it'll be processed by existing tooling (e.g. formatters).

I _do_ remember being annoyed that `repr` has no way of returning indented output.

Reply:
I'd prefer just switching it back. If you really want to debate it I'd rather debate it in a different PR so we can merge the rest of your changes sooner rather then later.